org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLDiagnostics.java

```diff
@@ -39,12 +39,13 @@ public XMLDiagnostics(XMLExtensionsRegistry extensionsRegistry) {
 		this.extensionsRegistry = extensionsRegistry;
 	}
 
-	public List<Diagnostic> doDiagnostics(DOMDocument xmlDocument, XMLValidationSettings validationSettings,
+	public DiagnosticsResult doDiagnostics(DOMDocument xmlDocument, XMLValidationSettings validationSettings,

```

Comment:
will it not break lemminx-maven?

Reply:
No,it willnot break lemminx-maven since they contribute with validation with IDiagnosticParticipant.